### PR TITLE
Near sdk collections docs update

### PR DESCRIPTION
**Pull request by deveusss**
Updated docs for Vector, TreeMap, LookupSet, LookupMap, UnorderedMap, UnorderedSet

### CODE REVIEW

File: near-sdk/src/collections/tree_map.rs [hunk 1]
```diff
@@ -130,6 +143,29 @@ where
...
+    /// 
+    /// assert_eq!(map.ceil_key(&5), Some(10));
+    /// assert_eq!(map.ceil_key(&10), Some(10));
+    /// assert_eq!(map.ceil_key(&11), Some(20));
+    /// assert_eq!(map.ceil_key(&20), Some(20));
+    /// assert_eq!(map.ceil_key(&49), Some(50));
+    /// assert_eq!(map.ceil_key(&50), Some(50));
+    /// assert_eq!(map.ceil_key(&51), None);
+    /// 
+    /// map.clear();
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:167]:
```suggestion
```

Don't really need this for this doc test

File: near-sdk/src/collections/tree_map.rs [hunk 2]
```diff
@@ -139,6 +175,25 @@ where
     }
 
     /// Returns the largest key that is less or equal to key given as the parameter
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m".to_vec());
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:182]:
```suggestion
    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m");
```

File: near-sdk/src/collections/tree_map.rs [hunk 3]
```diff
@@ -139,6 +175,25 @@ where
...
+    /// }
+    ///
+    /// assert_eq!(map.floor_key(&5), None);
+    /// assert_eq!(map.floor_key(&10), Some(10));
+    /// assert_eq!(map.floor_key(&11), Some(10));
+    /// assert_eq!(map.floor_key(&20), Some(20));
+    /// assert_eq!(map.floor_key(&49), Some(40));
+    /// assert_eq!(map.floor_key(&50), Some(50));
+    /// assert_eq!(map.floor_key(&51), Some(50));
+    /// map.clear();
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:195]:
```suggestion
```

File: near-sdk/src/collections/tree_map.rs [hunk 4]
```diff
@@ -153,6 +208,21 @@ where
     }
 
     /// Iterate entries in ascending order: given key (exclusive) to max (inclusive)
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m".to_vec());
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:215]:
```suggestion
    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m");
```

File: near-sdk/src/collections/tree_map.rs [hunk 5]
```diff
@@ -153,6 +208,21 @@ where
...
+    /// let one: Vec<u32> = vec![10, 20, 30, 40, 50,45, 35, 25, 15, 5];
+    /// for x in &one {
+    ///     map.insert(x, &42);
+    /// }
+    /// assert_eq!(
+    ///     map.iter_from(29).collect::<Vec<(u32, u32)>>(),
+    ///     vec![(30, 42), (35, 42), (40, 42), (45, 42), (50, 42)]
+    /// )
+    /// map.clear();
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:224]:
```suggestion
```

File: near-sdk/src/collections/tree_map.rs [hunk 6]
```diff
@@ -163,6 +233,21 @@ where
     }
 
     /// Iterate entries in descending order: given key (exclusive) to min (inclusive)
+    /// ///
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:236]:
```suggestion
    ///
```

File: near-sdk/src/collections/tree_map.rs [hunk 7]
```diff
@@ -163,6 +233,21 @@ where
...
+    /// # Examples
+    ///
+    /// ```
+    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m".to_vec());
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:240]:
```suggestion
    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m");
```

File: near-sdk/src/collections/tree_map.rs [hunk 8]
```diff
@@ -153,6 +208,21 @@ where
...
+    /// let one: Vec<u32> = vec![10, 20, 30, 40, 50,45, 35, 25, 15, 5];
+    /// for x in &one {
+    ///     map.insert(x, &42);
+    /// }
+    /// assert_eq!(
+    ///     map.iter_from(29).collect::<Vec<(u32, u32)>>(),
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:221]: optional: perhaps change this to iterating from 30 to show off the exclusive nature of the start index?

File: near-sdk/src/collections/tree_map.rs [hunk 9]
```diff
@@ -163,6 +233,21 @@ where
...
+    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m".to_vec());
+    /// let one: Vec<u32> = vec![10, 20, 30, 40, 50,45, 35, 25, 15, 5];
+    /// for x in &one {
+    ///     map.insert(x, &42);
+    /// }
+    /// assert_eq!(
+    ///     map.iter_rev_from(45).collect::<Vec<(u32, u32)>>(),
+    ///     vec![(40, 42), (35, 42), (30, 42), (25, 42), (20, 42), (15, 42), (10, 42), (5, 42)]
+    /// );
+    /// map.clear();
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:249]:
```suggestion
```

File: near-sdk/src/collections/tree_map.rs [hunk 10]
```diff
@@ -173,6 +258,25 @@ where
     ///
     /// Panics if range start > end.
     /// Panics if range start == end and both bounds are Excluded.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m".to_vec());
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:265]:
```suggestion
    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m");
```

File: near-sdk/src/collections/tree_map.rs [hunk 11]
```diff
@@ -173,6 +258,25 @@ where
...
+    /// let one: Vec<u32> = vec![10, 20, 30, 40, 50];
+    /// let two: Vec<u32> = vec![45, 35, 25, 15,
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:267]: You're missing a closing brace here. Wondering why the need for two vectors if just indicating which keys to insert

File: near-sdk/src/collections/tree_map.rs [hunk 12]
```diff
@@ -173,6 +258,25 @@ where
...
+    /// ```
+    /// let mut map: TreeMap<u32, u32> = TreeMap::new(b"m".to_vec());
+    /// let one: Vec<u32> = vec![10, 20, 30, 40, 50];
+    /// let two: Vec<u32> = vec![45, 35, 25, 15, 
+    /// for x in &one {
+    ///     map.insert(x, &0);
+    /// }
+    /// for x in &two {
+    ///     map.insert(x, &0);
+    ///
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:273]: you probably need a `}` here, the loop above is not closed

File: near-sdk/src/collections/tree_map.rs [hunk 13]
```diff
@@ -173,6 +258,25 @@ where
...
+    ///     map.insert(x, &0);
+    /// }
+    /// for x in &two {
+    ///     map.insert(x, &0);
+    /// 
+    /// assert_eq!(
+    ///     map.range((Bound::Included(20), Bound::Excluded(30))).collect::<Vec<(u32, u32)>>(),
+    ///     vec![(20, 0), (25, 0)]
+    /// );
+    /// map.clear();
```
Comment (austinabell) [inline near-sdk/src/collections/tree_map.rs:278]:
```suggestion
```

File: near-sdk/src/collections/unordered_map.rs [hunk 14]
```diff
@@ -42,6 +42,13 @@ impl<K, V> UnorderedMap<K, V> {
     }
 
     /// Create new map with zero elements. Use `prefix` as a unique identifier.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///use near_sdk::collections::LookupSet;
+    ///let mut set: LookupSet<u32> = LookupSet::new(b"m".to_vec());
```
Comment (austinabell) [inline near-sdk/src/collections/unordered_map.rs:50]:
```suggestion
    /// use near_sdk::collections::LookupSet;
    /// let mut set: LookupSet<u32> = LookupSet::new(b"m");
```

File: near-sdk/src/collections/unordered_set.rs [hunk 15]
```diff
@@ -27,6 +27,13 @@ impl<T> UnorderedSet<T> {
     }
 
     /// Create new map with zero elements. Use `id` as a unique identifier.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    ///use near_sdk::collections::UnorderedSet;
+    ///let mut set: UnorderedSet<u32> = UnorderedSet::new(b"m".to_vec());
```
Comment (austinabell) [inline near-sdk/src/collections/unordered_set.rs:35]:
```suggestion
    /// use near_sdk::collections::UnorderedSet;
    /// let mut set: UnorderedSet<u32> = UnorderedSet::new(b"m");
```

File: near-sdk/src/collections/vector.rs [hunk 16]
```diff
@@ -39,6 +39,13 @@ impl<T> Vector<T> {
     }
 
     /// Create new vector with zero elements. Use `id` as a unique identifier on the trie.
+    /// 
+    /// # Examples
+    ///
+    /// ```
+    ///use near_sdk::collections::Vector;
+    ///let mut set: Vector<u32> = Vector::new(b"m".to_vec());
```
Comment (austinabell) [inline near-sdk/src/collections/vector.rs:47]:
```suggestion
    /// use near_sdk::collections::Vector;
    /// let mut set: Vector<u32> = Vector::new(b"m");
```

**Review by austinabell — CHANGES_REQUESTED** [review]

File: near-sdk/src/collections/lookup_map.rs [hunk 17]
```diff
@@ -22,6 +22,21 @@ pub struct LookupMap<K, V> {
...
+    /// ```
+    /// use near_sdk::borsh::{self, BorshSerialize};
+    /// use near_sdk::collections::LookupMap;
+    /// use near_sdk::BorshStorageKey;
+
+    /// #[derive(BorshSerialize, BorshStorageKey)]
+    /// enum StorageKey {
+    ///     Records,
+    ///     UniqueValues,
+    /// }
```
[the remark this answers is not in the document]
Comment (thedeveus) [reply near-sdk/src/collections/lookup_map.rs:37]: sure, I will update to byte version)

File: near-sdk/src/collections/lookup_set.rs [hunk 18]
```diff
@@ -20,6 +20,13 @@ pub struct LookupSet<T> {
 
 impl<T> LookupSet<T> {
     /// Create a new map. Use `element_prefix` as a unique prefix for trie keys.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use near_sdk::collections::LookupSet;
+    /// let mut set: LookupSet<u32> = LookupSet::new(b"m".to_vec());
```
Comment (ChaoticTempest) [inline near-sdk/src/collections/lookup_set.rs:28]:
```suggestion
    /// let mut set: LookupSet<u32> = LookupSet::new(b"m");
```

File: near-sdk/src/store/lookup_map/mod.rs [hunk 19]
```diff
@@ -260,6 +294,17 @@ where
...
+    ///
+    /// # Example
+    /// ```
+    /// use near_sdk::store::LookupMap;
+    ///
+    /// let mut map = LookupMap::new(b"m");
+    /// map.insert(1, "a");
+    /// if let Some(x) = map.get_mut(&1) {
+    /// *x = "b";
+    /// assert_eq!(map[&1], "b");
```
Comment (ChaoticTempest) [inline near-sdk/src/store/lookup_map/mod.rs:306]:
```suggestion
    /// if let Some(x) = map.get_mut(&1) {
    ///     *x = "b";
    ///     assert_eq!(map[&1], "b");
    /// }
```

**Review by ChaoticTempest — COMMENTED** [review]
`near-sdk/.DS_Store` and `near-sdk/src/.DS_Store` seems unnecessary to be adding to the repo. Best to remove those before merging this PR. Aside from that, just some minor stuff still left, but I'll leave it to @austinabell to approve and merge

**Comment by austinabell** [comment]
It seems you are still making additions. Can you let us know when you are done so we can review once, or consider making new additions in a new PR to break it up?

**Comment by thedeveus** [comment]
> It seems you are still making additions. Can you let us know when you are done so we can review once, or consider making new additions in a new PR to break it up?
> 
> Thanks so much for adding all of these examples!

I think i finished with store collections, I am not sure if I need to add more examples in current collections

**Comment by austinabell** [comment]
Not all doc tests are passing, if you run `cargo test` you should be able to see which ones to fix :)

**Comment by deveusss** [comment]
> Not all doc tests are passing, if you run `cargo test` you should be able to see which ones to fix :)

fixed, thank you:)

**Comment by mikedotexe** [comment]
Thanks @deveusss for doing this. Looks like we have a merge conflict to clean up

**Comment by austinabell** [comment]
Is there anything I can do to help resolve conflicts and/or fix the CI? Seems like this PR has gone a bit stale

**Comment by deveusss** [comment]
> 
yes, there is just two different version of examples and need to choose one) I ll fix this

**Comment by mikedotexe** [comment]
Just updated the master and CI is running. we'd love to see this get in, and I'll ask @deveusss if everything is ready to go? I see there's a change requested currently

**Pull request merged**